docpipeline/generate.py

```diff
@@ -168,29 +163,33 @@ def process_blob(blob, credentials, devsite_template, xrefs):
     # Rename the output TOC file to be _toc.yaml to match the expected
     # format. As well, support both toc.html and toc.yaml
     try:
-        shutil.move(
-            site_api_path.joinpath("toc.yaml"), site_api_path.joinpath("_toc.yaml")
-        )
+        shutil.move(site_path.joinpath("toc.yaml"), site_path.joinpath("_toc.yaml"))
     except FileNotFoundError:
-        shutil.move(
-            site_api_path.joinpath("toc.html"), site_api_path.joinpath("_toc.yaml")
-        )
-
-    # Copy the xrefmap file to the output directory.
-    xrefmap = site_path.joinpath("xrefmap.yml")
-    shutil.copy(xrefmap, site_api_path)
+        shutil.move(site_path.joinpath("toc.html"), site_path.joinpath("_toc.yaml"))
 
     # Add the prettyprint class to code snippets
-    add_prettyprint(site_api_path)
+    add_prettyprint(site_path)
 
     log.success(f"Done building HTML for {blob.name}. Starting upload...")
 
     # Reuse the same docs.metadata file. The original docfx- prefix is an
     # command line option when uploading, not part of docs.metadata.
-    shutil.copy(metadata_path, site_api_path)
+    shutil.copy(metadata_path, site_path)
 
     # Use the input blob name as the name of the xref file to avoid collisions.
     # The input blob has a "docfx-" prefix; make sure to remove it.
+    xrefmap = site_path.joinpath("xrefmap.yml")
+    xrefmap_lines = xrefmap.read_text().splitlines()
+    # The baseUrl must start with a scheme and domain. With no scheme, docfx
+    # assumes it's a file:// link.
+    base_url = (
+        f"baseUrl: https://cloud.google.com/{metadata.language}/docs/reference/"
+        + f"{metadata.name}/latest/"
+    )
+    # Insert base_url after the YamlMime first line.
```

Comment:
It's possible this is fine - I've always put it *before* the YamlMime line. I realize that's weird, but it's what's always worked for me...

Reply:
They both seem to work. So, I think I'd prefer to put it below the YamlMime.

Reply:
Fine by me.